code/edge/src/androidTest/java/com/adobe/marketing/mobile/ConfigOverridesFunctionalTests.kt

```diff
@@ -0,0 +1,647 @@
+        {
+            "configOverrides" :{
+                "com_adobe_experience_platform": {
+                    "datasets": {
+                        "event": {
+                            "datasetId": "eventDatasetIdOverride"
+                        },
+                        "profile": {
+                            "datasetId": "profileDatasetIdOverride"
+                        }
+                    }
+                },
+                "com_adobe_analytics": {
+                    "reportSuites": [
+                        "rsid1",
+                        "rsid2",
+                        "rsid3"
+                    ]
+                },
+                "com_adobe_identity": {
+                    "idSyncContainerId": "1234567"
+                },
+                "com_adobe_target": {
+                    "propertyToken": "samplePropertyToken"
+                }
+            }
+        }
+        """.trim()
+
+        val expectedConfigOverridesMap = JSONObject(expectedConfigOverridesString)
+        val expectedXdmDataMap = JSONObject(expectedXdmDataString)
+        val expectedCustomDataMap = JSONObject(expectedCustomDataString)
+
+        val xdmData = mapOf("testXdmKey" to "testXdmValue")
+        val customData = mapOf("testCustomKey" to "testCustomValue")
+        val configOverrides: Map<String, Any> = mapOf(
+            "com_adobe_experience_platform" to mapOf(
+                "datasets" to mapOf(
+                    "event" to mapOf("datasetId" to "eventDatasetIdOverride"),
+                    "profile" to mapOf("datasetId" to "profileDatasetIdOverride")
+                )
+            ),
+            "com_adobe_analytics" to mapOf(
+                "reportSuites" to listOf("rsid1", "rsid2", "rsid3")
+            ),
+            "com_adobe_identity" to mapOf(
+                "idSyncContainerId" to "1234567"
+            ),
+            "com_adobe_target" to mapOf(
+                "propertyToken" to "samplePropertyToken"
+            )
+        )
+
+        val experienceEvent = ExperienceEvent.Builder()
+            .setXdmSchema(xdmData)
+            .setData(customData)
+            .setDatastreamConfigOverride(configOverrides)
+            .build()
+        Edge.sendEvent(experienceEvent, null)
+
+        // verify
+        mockNetworkService.assertAllNetworkRequestExpectations()
+        val resultRequests = mockNetworkService.getNetworkRequestsWith(
+            EXEDGE_INTERACT_URL_STRING,
+            HttpMethod.POST,
+            TIMEOUT_MILLIS
+        )
+        assertEquals(1, resultRequests.size.toLong())
+
+        val payloadJSONData = getPayloadJson(resultRequests[0])
+        val eventJSONData = payloadJSONData.getJSONArray("events").getJSONObject(0)
+        val metaJSONData = payloadJSONData.getJSONObject("meta")
+
+        JSONAsserts.assertExactMatch(expectedXdmDataMap, eventJSONData)
+        JSONAsserts.assertExactMatch(expectedCustomDataMap, eventJSONData)
```

Comment:
nit: For custom data, can you use `assertEqual()` here?

Reply:
I used assertExactMatch so that I dont need to extract "data" field and can pass the eventJSONData for the comparison. Any reason to use assertEqual?

Reply:
Ah, I see. For some reason I thought you were extracting the "xdm" and "data" separately. I think this is fine then.